addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -7392,6 +7392,35 @@ registry.ImageTools = ImageHandlerOption.extend({
             this._requestUserValueWidgets("remove_img_shape_opt")[0].enable();
         }
     },
+    /**
+     * @override
+     */
+    async _select(previewMode, widget) {
```

Comment:
Not really, it's because it doesn't work otherwise, if I put this in the function methods (e.g. at the end of `setImgShapeHoverEffect` it wouldn't work..

It's because of the `_refreshPublicWidgets` which is in the `_select` of the `website` option class (options.Class) .. This `_select` activates all the options associated to the snippet and then triggers a refresh (`_refreshPublicWidgets`). And it is this refresh which cancels the animation preview which would be triggered in the methods. So the only way to avoid canceling this animation is to do it after the `_select`.

So instead of:

- click the option
- start of the `_select`
- triggers methods (**preview animation**)
- refresh widget (that cancel the animation)
- end of the `_select`

we do:

- click the option
- start of the `_select`
- triggers methods
- refresh widget
- end of the `_select`
- **preview animation**

Reply:
I will add a comment to try to explain that clearly ;)